src/extractor/AssignmentRuleExtractor.ts

```diff
@@ -58,21 +58,30 @@ export class AssignmentRuleExtractor {
             `${matchingKey}.coding[0].system`,
             `${matchingKey}.coding[0].display`
           );
-          return assignmentRule;
+          return [assignmentRule];
         } else if (matchingKey.endsWith('Quantity') && isQuantity(matchingValue)) {
```

Comment:
I think we may need to update our idea of what a `Quantity` is based on these changes. The issue I'm noticing is that if there is some FSH defined like this:
```
Extension: NonUCUMQuant
* valueQuantity = http://some-url.org#foo "these are other units"
* valueQuantity.value = 12
```
And we run that through SUSHI, the resulting StructureDefinition has `code`, `system`, and `unit` set on `Extension.valueQuantity` via a `patternQuantity`, and `value` set on `Extension.valueQuantity.value` via a `patternDecimal`. Since this code is thinking that a `Quantity` must have `code` and `value`, the `patternQuantity` is not processed as a `Quantity` here, and ends up getting set with a caret rule: 
```
Extension: NonUCUMQuant
Id: NonUCUMQuant
* valueQuantity ^patternQuantity = $some-url#foo "these are other units"
* valueQuantity.value = 12
```
which doesn't seem right. Note that if you update `isQuantity` to include Quantities that don't have a `value`, I think there will need to be some changes further down, in places where it is assumed `value` exists.

Reply:
That's an excellent point. FSH allows for assigning to a `Quantity` without a value, so goFSH shouldn't be surprised when a patternQuantity doesn't have a value. I'll make this more flexible.